src/status_im/contexts/keycard/unblock/events.cljs

```diff
@@ -0,0 +1,89 @@
+(ns status-im.contexts.keycard.unblock.events
+  (:require [utils.re-frame :as rf]
+            [utils.security.core :as security]))
+
+(rf/reg-event-fx :keycard/unblock.pin-created
+ (fn [{:keys [db]} [pin]]
+   {:db (assoc-in db [:keycard :unblock :pin] pin)
+    :fx [[:dispatch
+          [:open-modal :screen/use-recovery-phrase-dark
+           {:on-success #(rf/dispatch [:keycard/unblock.phrase-entered %])}]]]}))
+
+(rf/reg-event-fx :keycard/unblock.phrase-entered
+ (fn [{:keys [db]} [{:keys [phrase]}]]
+   {:db (assoc-in db [:keycard :unblock :masked-phrase] phrase)
+    :fx [[:dispatch [:navigate-back]]
+         [:dispatch [:open-modal :screen/keycard.unblock.ready-to-unblock]]]}))
+
+(rf/reg-event-fx :keycard/unblock.generate-and-load-key
+ (fn [{:keys [db]}]
+   (let [{:keys [masked-phrase pin]} (get-in db [:keycard :unblock])]
+     {:fx [[:effects.keycard/generate-and-load-key
+            {:mnemonic   (security/safe-unmask-data masked-phrase)
+             :pin        pin
+             :on-success (fn []
+                           (rf/dispatch [:keycard/disconnect])
+                           (rf/dispatch [:navigate-back])
+                           (rf/dispatch [:open-modal :screen/keycard.unblock.success]))}]]})))
+
+;; third stage, import keys
+(rf/reg-event-fx :keycard/unblock.import-keys
+ (fn []
+   {:fx [[:dispatch
```

Comment:
Hi @flexsurfer, Thank you very much for your pr.

Should we create a function for this event and use it instead of dispatching this event?

Reply:
yes it's still needed but :keycard/unblock.init-card-or-import-keys should be fixed instead, thanks